language/move-lang/src/expansion/dependency_ordering.rs

```diff
@@ -46,44 +81,61 @@ enum DepType {
     Friend,
 }
 
+#[derive(Clone)]
+enum ModuleIdentOrScriptName {
+    Module(ModuleIdent),
+    Script(String),
```

Comment:
Script names are not guaranteed to be unique. If you are going to do this, we should chat about adding that

Reply:
> Script names are not guaranteed to be unique.

Thanks for spotting this! I did not realize that. Will change as you suggested.

Reply:
It might be something we want to do though. It has been a bit of a pain for the CLI.

I don't really love the syntax if we end up doing that, but that is a discussion for another time

Reply:
Ahh, sorry that I was confused before.

I think the `Script(String)` here should be unique for all scripts in the context. The reason is that the `scripts` are taken from the `keyed-script-name` transformation
```rust
let mut collected: BTreeMap<String, Vec<E::Script>> = BTreeMap::new();
...
let mut keyed: BTreeMap<String, E::Script> = BTreeMap::new();
...
```
where, if I read the code correctly, scripts with the same name will be keyed into `<name>_0`, `<name>_1`, ... `<name>_n`.

We should be safe here but I'll rename the enum to `ModuleIdentOrKeyedScriptName` to reflect that we are not using the plain script name as key.

Reply:
I guess I forgot I even did this. 

So we are safe but I still am somehow bothered by the name here. I think it just feels real weird to have an enum that is `AorB { A, B }`. Feels like there should be a better name there somehow. 

Maybe just `NodeIdent` since it is just used by this module